src/main/java/core/basesyntax/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.transaction.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> converterToTransaction(List<String> inputData) {
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (String data: inputData) {
+            if (data.equals("type,fruit,quantity")) {
+                continue;
+            }
+            String[] parts = data.split(",");
+            String operationCode = parts[0].trim();
+            String fruitName = parts[1].trim();
+            int quantity = Integer.parseInt(parts[2].trim());
+
+            // Determine the type of activity
+            FruitTransaction.Operation operation
+                    = FruitTransaction.Operation.fromCode(operationCode);

```

Comment:
You are not handling the case where the operation code might be invalid. What if the data contains a code that does not correspond to any operation? This can lead to a NullPointerException when you try to call the fromCode() method. Consider adding a check for the validity of the operation code before calling this method.